ingests_tracker/src/test/scala/uk/ac/wellcome/platform/storage/ingests_tracker/services/MessagingServiceTest.scala

```diff
@@ -0,0 +1,224 @@
+    }
+
+  describe("Ingest with status Succeeded") {
+    val successfulIngest = createIngestWith(status = Succeeded)
+
+    val callbackNotification = CallbackNotification(
+      ingestId = successfulIngest.id,
+      callbackUri = successfulIngest.callback.get.uri,
+      payload = successfulIngest
+    )
+
+    withMessagingService() {
+      case (callbackSender, ingestsSender, messagingService) =>
+        val sent = messagingService.send(successfulIngest)
+
+        it("returns a Success") {
+          sent shouldBe a[Success[_]]
+        }
+
+        it("sends a CallbackNotification") {
+          callbackSender.getMessages[CallbackNotification] shouldBe Seq(
+            callbackNotification
+          )
+        }
+
+        it("sends on the Ingest") {
+          ingestsSender.getMessages[Ingest] shouldBe Seq(successfulIngest)
+        }
+    }
+  }
+
+  describe("Ingest with status Failed") {
+    val failedIngest = createIngestWith(status = Failed)
+
+    val callbackNotification = CallbackNotification(
+      ingestId = failedIngest.id,
+      callbackUri = failedIngest.callback.get.uri,
+      payload = failedIngest
+    )
+
+    withMessagingService() {
+      case (callbackSender, ingestsSender, messagingService) =>
+        val sent = messagingService.send(failedIngest)
+
+        it("returns a Success") {
+          sent shouldBe a[Success[_]]
+        }
+
+        it("sends a CallbackNotification") {
+          callbackSender.getMessages[CallbackNotification] shouldBe Seq(
+            callbackNotification
+          )
+        }
+
+        it("sends on the Ingest") {
+          ingestsSender.getMessages[Ingest] shouldBe Seq(failedIngest)
+        }
+    }
+  }
+
+  describe("Ingest with status Accepted") {
+    val acceptedIngest = createIngestWith(status = Accepted)
+
+    withMessagingService() {
+      case (callbackSender, ingestsSender, messagingService) =>
+        val sent = messagingService.send(acceptedIngest)
+
+        it("returns a Success") {
+          sent shouldBe a[Success[_]]
+        }
+
+        it("does not send a CallbackNotification") {
+          callbackSender.getMessages[CallbackNotification] shouldBe empty
+        }
+
+        it("sends on the Ingest") {
+          ingestsSender.getMessages[Ingest] shouldBe Seq(acceptedIngest)
+        }
+    }
+  }
+
+  describe("Ingest with status Processing") {
+    val processingIngest = createIngestWith(status = Processing)
+
+    withMessagingService() {
+      case (callbackSender, ingestsSender, messagingService) =>
+        val sent = messagingService.send(processingIngest)
+
+        it("returns a Success") {
+          sent shouldBe a[Success[_]]
+        }
+
+        it("does not send a CallbackNotification") {
+          callbackSender.getMessages[CallbackNotification] shouldBe empty
+        }
+
+        it("sends on the Ingest") {
+          ingestsSender.getMessages[Ingest] shouldBe Seq(processingIngest)
+        }
+    }
+  }
```

Comment:
I wonder if these tests could be turned into table-driven tests to reduce the repetition, like the NotificationService tests.